dss/api/files.py

```diff
@@ -112,6 +114,20 @@ def get_helper(uuid: str, replica: Replica, version: str=None, token: str=None):
 def _verify_checkout(
         replica: Replica, token: typing.Optional[str], file_metadata: dict, blob_path: str,
 ) -> typing.Tuple[str, bool]:
+    try:
+        cloud_handle = Config.get_blobstore_handle(replica)
+        hca_handle = Config.get_hcablobstore_handle(replica)
+        last_modified = cloud_handle.get_last_modified_date(replica.checkout_bucket, "files/{}".format(blob_path))
+        stale_date = datetime.datetime.now(pytz.utc) - datetime.timedelta(days=int(os.environ['DSS_BLOB_PUBLIC_TTL']))
+
+        if ((last_modified > stale_date) and
+            (hca_handle.verify_blob_checksum_from_dss_metadata(replica.checkout_bucket,
+                                                               get_dst_key(blob_path),
+                                                               file_metadata))):
+            return "", True
```

Comment:
Would consider making the str optional for the return type to make this less awkward.

Reply:
Also, if the file is present but the checksum does not match, I think it's worth logging some very serious error.

Reply:
How would one make the return type Tuple(str, bool) or bool?